src/crossql/Migrations/MigrationRunner.cs

```diff
@@ -20,182 +19,230 @@ public MigrationRunner(IDbProvider dbProvider)
+                        failedStep = MigrationStep.ClientFinish;
+                        await migration.ClientFinishMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ServerSetup:
+                        failedStep = MigrationStep.ServerSetup;
+                        await migration.ServerSetupMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ServerMigrate:
+                        failedStep = MigrationStep.ServerMigrate;
+                        await migration.ServerMigrate(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    case MigrationStep.ServerFinish:
+                        failedStep = MigrationStep.ServerFinish;
+                        await migration.ServerFinishMigration(database, dbProvider).ConfigureAwait(false);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(migrationStep), migrationStep, $"An invalid {nameof(MigrationStep)} was passed to the {nameof(MigrationRunner)}");
+                }
 
-        private async Task<DatabaseVersionModel> GetMigrationInformationAsync(IDbMigration migration)
+                var dbCommand = database.ToString();
+                if (!string.IsNullOrWhiteSpace(dbCommand))
+                    await _dbProvider.ExecuteNonQuery(dbCommand);
+            }
+            catch (Exception ex)
+            {
+                var database = new Database(dbProvider.DatabaseName, dbProvider.Dialect);
+                await migration.MigrationFailed(database, dbProvider, failedStep, ex);
+                throw;
+            }
+        }
+
+        private async Task<Version> GetMigrationInformationAsync(IDbMigration migration)
         {
-            var databaseVersions = await _dbProvider.Query<DatabaseVersionModel>()
+            var databaseVersion = await _dbProvider.Query<Version>()
                 .Where(dbv => dbv.VersionNumber == migration.MigrationVersion)
                 .OrderBy(v => v.VersionNumber, OrderDirection.Descending)
-                .Select()
+                .FirstOrDefaultAsync()
                 .ConfigureAwait(false);
-            var databaseVersion = databaseVersions.FirstOrDefault();
 
-            if (databaseVersion == null)
+            if (databaseVersion != null) return databaseVersion;
+
+            databaseVersion = new Version
             {
-                databaseVersion = new DatabaseVersionModel
-                {
-                    VersionNumber = migration.MigrationVersion,
-                    MigrationDate = DateTime.UtcNow
-                };
-                await _dbProvider.Create(databaseVersion).ConfigureAwait(false);
-            }
+                VersionNumber = migration.MigrationVersion,
+                MigrationDate = DateTimeOffset.Now
```

Comment:
Should we be using `DateTimeOffset.Now` or `DateTimeOffset.UtcNow`?

Reply:
note: this really doesn't matter all that much since it's not a table that's designed to sync.